client/blocks/signup-form/index.jsx

```diff
@@ -246,87 +246,94 @@ class SignupForm extends Component {
 		] );
 
 		const data = mapKeys( pick( fields, fieldsForValidation ), ( value, key ) => snakeCase( key ) );
-		wpcom.undocumented().validateNewUser( data, ( error, response ) => {
-			if ( this.props.submitting ) {
-				// this is a stale callback, we have already signed up or are logging in
-				return;
-			}
-
-			if ( error || ! response ) {
-				return debug( error || 'User validation failed.' );
-			}
-
-			let messages = response.success
-				? {}
-				: mapKeys( response.messages, ( value, key ) => camelCase( key ) );
-
-			// Prevent "field is empty" error messages from displaying prematurely
-			// before the form has been submitted or before the field has been interacted with (is dirty).
-			if ( ! this.state.submitting ) {
-				messages = this.filterUntouchedFieldErrors( messages );
-			}
-
-			forEach( messages, ( fieldError, field ) => {
-				if ( ! formState.isFieldInvalid( this.state.form, field ) ) {
+		wpcom.req.post(
+			'/signups/validation/user',
```

Comment:
Nit: I've noticed that in the `undocumented` request, we have a trailing slash. Here, we don't. I think having a trailing slash was a mistake in the first place.

Reply:
Yes, I removed the trailing slash intentionally. The REST endpoint doesn't really need it.